packages/adapters/ethers/src/client.ts

```diff
@@ -631,9 +631,12 @@ export class EthersAdapter {
       if (provider) {
         const { addresses, chainId } = await EthersHelpersUtil.getUserInfo(provider)
         const firstAddress = addresses?.[0]
-        const caipAddress = `${this.chainNamespace}:${chainId}:${firstAddress}` as CaipAddress
+        const caipNetwork = this.caipNetworks.find(c => c.id === chainId) ?? this.caipNetworks[0]
+        const caipAddress =
+          `${this.chainNamespace}:${caipNetwork?.id}:${firstAddress}` as CaipAddress
```

Comment:
why not use the caipNetwork id directly?

Reply:
No, I meant `${caipNetwork.namespace}:${caipNetwork?.id}:${firstAddress}` or `${caipNetwork.caipNetworkId}:${firstAddress}`

(unsure about proper names but you get what I mean)